### PR TITLE
Ports "Bluespace research updates" and "Teleportation standardization, channels, and interception"

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Ports /tg/station PRs #40245, #41371 and also #40599, #41694, #41879, #44602 and #44664, that should be it.
Also personally made clock cult use do_teleport where appliable, with forced on true due to rebee being a noteleport area

## Why It's Good For The Game
Keeping this repository less out of date from the once was upstream.
And quantum pad keycards feature.
I'll get on the anti-megafauna teleportation later maybe.

## Changelog
:cl: Ghommie (original PRs by XDTM, optimumtact, Nichlas0010 and monster860)
add: Added Quantum Keycards, devices that can link to a quantum pad, and can be used on any other quantum pad to teleport to its linked pad.
spellchecking: Renamed "Bluespace Teleportation Tech" tech node to "Bluespace Travel".
tweak: Moved roasting sticks from the "Bluespace Travel" to "Practical Bluespace".
add: Spraying holy water on tiles will now prevent cult-based teleportation from using them as a destination point.
tweak: Quantum, wormhole and magic teleportation is no longer disrupted by bags of holding.
fix: You are now also blocked from teleporting IN to no-teleport areas, not just out of them.
tweak: Quantum teleportation now makes pretty rainbow sparks instead of the normal ones.
fix: Non-bluespace teleportation (spells etc.) no longer makes sparks.
fix: Fixed teleportation deleting mob spawners like golem shells and ashwalker eggs.
/:cl:

### CODE REVIEW

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -227,33 +227,42 @@
...
-	design_ids = list("bluespace_matter_bin", "femto_mani", "triphasic_scanning", "bluespace_crystal")
-	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 10000)
-	export_price = 5000
-
 /datum/techweb_node/bluespace_power
 	id = "bluespace_power"
 	display_name = "Bluespace Power Technology"
 	description = "Even more powerful.. power!"
-	prereq_ids = list("adv_power", "adv_bluespace")
+	prereq_ids = list("adv_power", "practical_bluespace")
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm]: keep this adv bluespace

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -227,33 +227,42 @@
...
+	design_ids = list("beacon", "xenobioconsole", "telesci_gps", "bluespace_crystal", "xenobio_monkeys")
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
 	export_price = 5000
 
+/datum/techweb_node/bluespace_travel
+	id = "bluespace_travel"
+	display_name = "Bluespace Travel"
+	description = "Application of Bluespace for static teleportation technology."
+	prereq_ids = list("practical_bluespace")
+	design_ids = list("tele_station", "tele_hub", "quantumpad", "launchpad", "launchpad_console", "bluespace_pod")
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm]: Already a node, see warping tech

File: code/modules/research/techweb/all_nodes.dm [hunk 3]
```diff
@@ -227,33 +227,42 @@
 	display_name = "Basic Bluespace Theory"
 	description = "Basic studies into the mysterious alternate dimension known as bluespace."
 	prereq_ids = list("base", "datatheory")
-	design_ids = list("beacon", "xenobioconsole", "telesci_gps")
+	design_ids = list("beacon", "xenobioconsole", "telesci_gps", "bluespace_crystal", "xenobio_monkeys")
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm]: PR https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8660
#8660 handles the disks

File: code/modules/research/techweb/all_nodes.dm [hunk 4]
```diff
@@ -227,33 +227,42 @@
...
+	design_ids = list("tele_station", "tele_hub", "quantumpad", "launchpad", "launchpad_console", "bluespace_pod")
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 5000)
+	export_price = 5000
+
+/datum/techweb_node/micro_bluespace
+	id = "micro_bluespace"
+	display_name = "Miniaturized Bluespace Research"
+	description = "Extreme reduction in space required for bluespace engines, leading to portable bluespace technology."
+	prereq_ids = list("bluespace_travel", "practical_bluespace", "high_efficiency")
+	design_ids = list("bluespace_matter_bin", "femto_mani", "triphasic_scanning", "bag_holding", "quantum_keycard", "wormholeprojector", "xenobio_slimeadv")
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm]: Place the keycard in warping tech as thats what its used for. Dont add boh or worm hole into a early game tech like this.

File: code/modules/research/techweb/all_nodes.dm [hunk 5]
```diff
@@ -227,33 +227,42 @@
...
 	id = "practical_bluespace"
 	display_name = "Applied Bluespace Research"
 	description = "Using bluespace to make things faster and better."
 	prereq_ids = list("bluespace_basic", "engineering")
-	design_ids = list("bs_rped","biobag_holding","minerbag_holding", "bluespacebeaker", "bluespacesyringe", "phasic_scanning", "bluespacesmartdart")
+	design_ids = list("bs_rped","biobag_holding","minerbag_holding", "bluespacebeaker", "bluespacesyringe", "phasic_scanning", "bluespacesmartdart", "xenobio_slimebasic")
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 5000)
 	export_price = 5000
 
-/datum/techweb_node/adv_bluespace
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm]: Dont remove this

File: code/modules/research/techweb/all_nodes.dm [hunk 6]
```diff
@@ -227,33 +227,42 @@
...
+/datum/techweb_node/bluespace_travel
+	id = "bluespace_travel"
+	display_name = "Bluespace Travel"
+	description = "Application of Bluespace for static teleportation technology."
+	prereq_ids = list("practical_bluespace")
+	design_ids = list("tele_station", "tele_hub", "quantumpad", "launchpad", "launchpad_console", "bluespace_pod")
+	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 5000)
+	export_price = 5000
+
+/datum/techweb_node/micro_bluespace
```
Comment (Trilbyspaceclone) [inline code/modules/research/techweb/all_nodes.dm]: Remove this keep it as adv blue space rather then snowflake micro - It would be femto

**Review by Trilbyspaceclone — CHANGES_REQUESTED** [review]
Please dont just TG are tech webs and make shit easy/early game like this.
Your really throwing of the balances

File: code/modules/research/techweb/all_nodes.dm (hunk 3 above)
Comment (Ghommie) [reply code/modules/research/techweb/all_nodes.dm]: It'll still conflict anyway.

Comment (Ghommie) [reply code/modules/research/techweb/all_nodes.dm]: or not.

**Review by silicons — APPROVED** [review]

**Pull request merged**